google/cloud/error_reporting/_gapic.py

```diff
@@ -62,7 +61,10 @@ def report_error_event(self, error_report):
             Use
             :meth:~`google.cloud.error_reporting.client._build_error_report`
         """
-        project_name = self._gapic_api.project_path(self._project)
-        error_report_payload = report_errors_service_pb2.ReportedErrorEvent()
-        ParseDict(error_report, error_report_payload)
-        self._gapic_api.report_error_event(project_name, error_report_payload)
+        project_name = f"projects/{self._project}"
+        error_report_payload = google.cloud.errorreporting_v1beta1.ReportedErrorEvent.from_json(
```

Comment:
Is it possible to pass the dict directly to `ReportedErrorEvent`? https://github.com/googleapis/proto-plus-python/blob/master/proto/message.py#L366 or does that result in an error?

@software-dov could you take a quick look?

Reply:
It is possible to go straight from dict to proto-plus message withouth routing through json.
```python
error_reporting_payload = google.cloud.errorreporting_v1beta1.ReportedErrorEvent(**error_report)

Reply:
Let me try it. I used json because `error_report` uses camel case keys like `serviceAccount`, but `ReportedErrorEvent` uses snake case keys like `service_account`.

Reply:
Oy. If that winds up causing problems, use the following utility function
```python
def split_caps_and_downcase(s):
    # Assuming s is CamelCase or similar.
    toks = []
    base_idx = 0
    for idx, char in enumerate(s):
        if base_idx != idx and char.isupper():
            toks.append(s[base_idx:idx].lower())
            base_idx = idx

    toks.append(s[base_idx:].lower())
    return toks

ReportedErrorEvent(**{"_".join(split_caps_and_downcase(k)) for k, v in error_report.items()})
```

Reply:
@arithmetic1728 I think the solution you have here (routing through json) is preferable to adding another function. Could you add a comment explaining the conversion?

Reply:
I tested `ReportedErrorEvent(**error_report)`, still get the keyError. Sure, I will add a comment explaining the conversion.

Reply:
comments added